lib/jar.ts

```diff
@@ -0,0 +1,22 @@
+import * as path from 'path';
+import * as crypto from 'crypto';
+import * as fs from 'fs';
+
+export function getTempPomPathForJar(root: string, jar: string): string {
+  const sha1 = getSha1ForJar(path.join(root, jar));
+
+  // return path to root of pom.xml
+  return '';
+}
+
+export function isJar(file: string) {
+  return file.match(/\.(([jw]ar)|(zip))$/);
+}
+
+export function getSha1ForJar(filePath) {
+  const fileContents = fs.readFileSync(filePath);
```

Comment:
try/catch handling the case where file does not exist or there is a problem reading? add test for this

Reply:
added above

Reply:
don't export this function if the validation is done there?

Reply:
readFileSync could still error is process does not have permission to read an existing file - probably best to move try/catch here?